--- C:/HomServ/home/second/phpsvnclient-105.php	Thu Jul 21 15:37:58 2011
+++ C:/HomServ/home/second/phpsvnclient-107.php	Thu Jul 21 15:38:10 2011
@@ -133,8 +133,9 @@
     public $errNro;
     private $storeDirectoryFiles = array();
     private $lastDirectoryFiles;
     public $actVersion;
+    private $path_exec_after_completition = '';
 
     public function phpsvnclient($url = 'http://phpsvnclient.googlecode.com/svn/', $user = false, $pass = false) {
 	$this->__construct($url, $user, $pass);
 	register_shutdown_function(array(&$this, '__destruct'));
@@ -152,17 +153,16 @@
 	$this->actVersion = $this->getVersion();
     }
 
     function createDirs($path) {
-        $createDir='';
-        $dirs = explode("/", $path);
+	$dirs = explode("/", $path);
 
-        foreach ($dirs as $dir) {
-            if ($dir != "") {
-                $createDir .= '/'.$dir;
-                @mkdir($createDir);
-            }
-        }
+	foreach ($dirs as $dir) {
+	    if ($dir != "") {
+		$createDir = substr($path, 0, strpos($path, $dir) + strlen($dir));
+		@mkdir($createDir);
+	    }
+	}
     }
 
     function removeDirs($path) {
 	if (is_dir($path)) {
@@ -233,8 +233,11 @@
 		$hOut = fopen($createPath, 'w');
 		fwrite($hOut, $contents);
 		fclose($hOut);
 	    }
+	}
+	if ($this->path_exec_after_completition != '') {
+	    $this->exec_after_completition();
 	}
     }
 
     /**
@@ -331,8 +334,12 @@
 		$hOut = fopen($outPath . '/.svn/entries', 'w');
 		fwrite($hOut, $this->actVersion);
 		fclose($hOut);
 	    }
+	}
+
+	if ($this->path_exec_after_completition != '') {
+	    $this->exec_after_completition();
 	}
     }
 
     /**
@@ -872,8 +879,16 @@
      *  @access private
      */
     private function cleanURL($url) {
 	return preg_replace("/((^:)\/\/)/", "//", $url);
+    }
+
+    function exec_after_completition() {
+	require_once $this->path_exec_after_completition;
+    }
+
+    function set_job_for_exec_after_completition($path_to_file) {
+	$this->path_exec_after_completition = $path_to_file;
     }
 
 }
 
