--- C:/HomServ/home/second/phpsvnclient-107.php	Thu Jul 21 15:38:10 2011
+++ C:/HomServ/home/second/phpsvnclient-105.php	Thu Jul 21 15:37:58 2011
@@ -134,7 +134,6 @@
     private $storeDirectoryFiles = array();
     private $lastDirectoryFiles;
     public $actVersion;
-    private $path_exec_after_completition = '';
 
     public function phpsvnclient($url = 'http://phpsvnclient.googlecode.com/svn/', $user = false, $pass = false) {
 	$this->__construct($url, $user, $pass);
@@ -154,14 +153,15 @@
     }
 
     function createDirs($path) {
-	$dirs = explode("/", $path);
+        $createDir='';
+        $dirs = explode("/", $path);
 
-	foreach ($dirs as $dir) {
-	    if ($dir != "") {
-		$createDir = substr($path, 0, strpos($path, $dir) + strlen($dir));
-		@mkdir($createDir);
-	    }
-	}
+        foreach ($dirs as $dir) {
+            if ($dir != "") {
+                $createDir .= '/'.$dir;
+                @mkdir($createDir);
+            }
+        }
     }
 
     function removeDirs($path) {
@@ -235,9 +235,6 @@
 		fclose($hOut);
 	    }
 	}
-	if ($this->path_exec_after_completition != '') {
-	    $this->exec_after_completition();
-	}
     }
 
     /**
@@ -336,10 +333,6 @@
 		fclose($hOut);
 	    }
 	}
-
-	if ($this->path_exec_after_completition != '') {
-	    $this->exec_after_completition();
-	}
     }
 
     /**
@@ -880,14 +873,6 @@
      */
     private function cleanURL($url) {
 	return preg_replace("/((^:)\/\/)/", "//", $url);
-    }
-
-    function exec_after_completition() {
-	require_once $this->path_exec_after_completition;
-    }
-
-    function set_job_for_exec_after_completition($path_to_file) {
-	$this->path_exec_after_completition = $path_to_file;
     }
 
 }
